django/contrib/sitemaps/views.py

```diff
@@ -65,16 +67,26 @@ def sitemap(request, sitemaps, section=None,
                 site = site()
             urls.extend(site.get_urls(page=page, site=req_site,
                                       protocol=req_protocol))
+            if all_sites_lastmod:
+                site_lastmod = getattr(site, 'latest_lastmod', None)
+                if site_lastmod is not None:
+                    if type(site_lastmod) is datetime.date:
+                        site_lastmod = datetime.datetime(site_lastmod.year,
+                                                         site_lastmod.month,
+                                                         site_lastmod.day)
```

Comment:
Maybe you could convert `datetime` to `utctimetuple()` and `date` to `timetuple()` here instead. That's the actual type you'll need for `timegm` anyway and you won't have to deal with naive and aware datetime comparison this way.